### PR TITLE
hw-26.1

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

File: src/main/java/application/controller/AddProductController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/products/add")
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("application");
+    private final ProductService productService =
```
Comment (dariiabutorina) [inline src/main/java/application/controller/AddProductController.java]:
```suggestion
    private ProductService productService =
```

Comment (dariiabutorina) [reply src/main/java/application/controller/AddProductController.java]: I was told to remove all the 'final' modifiers of the service objects.

File: src/main/java/application/controller/AddProductController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/view/product/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        long price = Long.parseLong(req.getParameter("price"));
```
Comment (dariiabutorina) [inline src/main/java/application/controller/AddProductController.java:29]: We should check whether the String is not empty.

File: src/main/java/application/controller/DeleteUserController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("id"));
+        userService.delete(id);
+        resp.sendRedirect(req.getContextPath() + "/user/all");
+
+    }
```
Comment (dariiabutorina) [inline src/main/java/application/controller/DeleteUserController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/user/all");
    }
```

File: src/main/java/application/controller/GetShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/myCart")
+public class GetShoppingCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("application");
+    private ShoppingCartService shoppingCartService
+            = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
```
Comment (dariiabutorina) [inline src/main/java/application/controller/GetShoppingCartController.java]:
```suggestion
    private ShoppingCartService shoppingCartService =
            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
```

File: src/main/java/application/controller/RegisterController.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        req.getRequestDispatcher("/WEB-INF/view/user/registration.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        String repeatPassword = req.getParameter("repeat");
```
Comment (dariiabutorina) [inline src/main/java/application/controller/RegisterController.java:34]: Strings' length check

File: src/main/java/application/controller/DeleteUserController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    private static final Injector injector = Injector.getInstance("application");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("id"));
+        userService.delete(id);
+        resp.sendRedirect(req.getContextPath() + "/user/all");
+
```
Comment (DmGrt) [inline src/main/java/application/controller/DeleteUserController.java]: remove:)

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Add the method that deletes products from the cart, please

File: src/main/webapp/WEB-INF/view/product/add.jsp [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Update product list</title>
+</head>
+<body>
+
```
Comment (DmGrt) [inline src/main/webapp/WEB-INF/view/product/add.jsp]: same:)

File: src/main/webapp/WEB-INF/view/product/add.jsp [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+<form method="post" action="${pageContext.request.contextPath}/products/add">
+    Name <input type="text" name="name" required>
+    Price <input type="text" name="price" required>
+    <button type="submit">Add</button>
+</form>
+
```
Comment (DmGrt) [inline src/main/webapp/WEB-INF/view/product/add.jsp]: same

File: src/main/java/application/controller/AddProductController.java (hunk 2 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/AddProductController.java:29]: I've made this check in jsp file

File: src/main/webapp/web.xml [hunk 9]
```diff
@@ -3,12 +3,5 @@
...
          version="3.1">
-    <servlet>
-        <servlet-name>index</servlet-name>
-        <servlet-class>application.controllers.IndexController</servlet-class>
-    </servlet>
-    <servlet-mapping>
-        <servlet-name>index</servlet-name>
-        <url-pattern>/index</url-pattern>
-    </servlet-mapping>
+
```
Comment (DmGrt) [inline src/main/webapp/web.xml]: one more

File: src/main/java/application/controller/RegisterController.java (hunk 5 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/RegisterController.java:34]: Same. The check is in jsp

**Review by DmGrt — COMMENTED** [review]
Great job! just let's remove some empty lines)

File: src/main/java/application/controller/AddToCartController.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import application.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/addToCart")
+public class AddToCartController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/application/controller/AddToCartController.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

File: src/main/java/application/controller/AddToCartController.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+import application.service.ProductService;
+import application.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/addToCart")
```
Comment (Sofasmile) [inline src/main/java/application/controller/AddToCartController.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/application/controller/DeleteUserController.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import application.lib.Injector;
+import application.service.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/delete")
```
Comment (Sofasmile) [inline src/main/java/application/controller/DeleteUserController.java]: do we delete `user` or `product` or `shoppingCart`?

File: src/main/java/application/controller/GetShoppingCartController.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+import application.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/myCart")
```
Comment (Sofasmile) [inline src/main/java/application/controller/GetShoppingCartController.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/application/controller/DeleteUserController.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("application");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("id"));
+        userService.delete(id);
+        resp.sendRedirect(req.getContextPath() + "/user/all");
```
Comment (Sofasmile) [inline src/main/java/application/controller/DeleteUserController.java:28]: let's remove shopping cart too

File: src/main/java/application/controller/AddToCartController.java (hunk 11 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/AddToCartController.java]: done

File: src/main/java/application/controller/AddToCartController.java (hunk 10 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/AddToCartController.java]: done

File: src/main/java/application/controller/DeleteUserController.java (hunk 12 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/DeleteUserController.java]: done

File: src/main/java/application/controller/DeleteUserController.java (hunk 14 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/DeleteUserController.java:28]: done

File: src/main/java/application/controller/GetShoppingCartController.java (hunk 13 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/GetShoppingCartController.java]: done

File: src/main/java/application/controller/AddProductToCartController.java [hunk 15]
```diff
@@ -10,8 +10,8 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebServlet("/addToCart")
-public class AddToCartController extends HttpServlet {
+@WebServlet("/cart/addProduct")
```
Comment (Sofasmile) [inline src/main/java/application/controller/AddProductToCartController.java]: /cart/addProduct - bad naming https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html
/cart/add-product or /cart/product/add - good naming

File: src/main/java/application/controller/DeleteUserController.java [hunk 16]
```diff
@@ -9,16 +10,19 @@
...
     private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         Long id = Long.valueOf(req.getParameter("id"));
         userService.delete(id);
+        shoppingCartService.delete(id);
```
Comment (Sofasmile) [inline src/main/java/application/controller/DeleteUserController.java]: we remove ShoppingCart by userId, but method delete() expect shoppingCartId

File: src/main/java/application/controller/AddProductToCartController.java (hunk 15 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/AddProductToCartController.java]: done

File: src/main/java/application/controller/DeleteUserController.java (hunk 16 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/DeleteUserController.java]: done

File: src/main/java/application/controller/DeleteUserController.java [hunk 17]
```diff
@@ -22,7 +23,8 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         Long id = Long.valueOf(req.getParameter("id"));
         userService.delete(id);
-        shoppingCartService.delete(id);
+        ShoppingCart cart = shoppingCartService.get(id);
```
Comment (Sofasmile) [inline src/main/java/application/controller/DeleteUserController.java]: давай перейменуємо метод get() на getByUserId() Якщо він дістає корзину за ід юзера

Comment (julia-shvabovich) [reply src/main/java/application/controller/DeleteUserController.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**